packages/tx/src/eip1559Transaction.ts

```diff
@@ -202,13 +205,10 @@ export default class FeeMarketEIP1559Transaction extends BaseTransaction<FeeMark
       toBuffer(maxPriorityFeePerGas === '' ? '0x' : maxPriorityFeePerGas)
     )
 
-    this._validateCannotExceedMaxInteger(
-      {
-        maxFeePerGas: this.maxFeePerGas,
-        maxPriorityFeePerGas: this.maxPriorityFeePerGas,
-      },
-      256
-    )
+    this._validateCannotExceedMaxInteger({
+      maxFeePerGas: this.maxFeePerGas,
+      maxPriorityFeePerGas: this.maxPriorityFeePerGas,
+    })
```

Comment:
Should the fee market tx also not have a check to ensure that `gasLimit * maxFeePerGas` does not overflow the `MAX_UINT`?

Reply:
Good catch.  I'll add that here.

Reply:
So on this one, if I add a check in the `_validateCannotExceedMaxInteger` function call in the 1559 tx constructor, [this test](https://github.com/ethereumjs/ethereumjs-monorepo/blob/531e4b3a1843833b3015ab185fc610f4f2da5d67/packages/tx/test/eip1559.spec.ts#L169) now fails.  From looking at it, I think this test is actually invalid since maxFeePerGas of `TWO_POW256.subn(1)` (aka `MAX_INTEGER`) x gasLimit of 100 would produce a value greater than `MAX_INTEGER`, right?

Reply:
That's what the test is indicating so I've made the associated changes and pushed but just want to confirm I'm not imaging things.

Reply:
I think using the function with "gasLimitXmaxFeePerGas" is a little confusing, can you just do the math yourself in a one liner like the legacy gasPrice * gasLimit?